cluster/src/main/java/com/hartwig/pipeline/execution/vm/ComputeEngine.java

```diff
@@ -42,48 +41,56 @@
 import net.jodah.failsafe.Failsafe;
 import net.jodah.failsafe.RetryPolicy;
 
-public class ComputeEngine implements CloudExecutor<VirtualMachineJobDefinition> {
+public class ComputeEngine {
     private static final int NUMBER_OF_375G_LOCAL_SSD_DEVICES = 4;
     private final static String APPLICATION_NAME = "vm-hosted-workload";
     static final String ZONE_EXHAUSTED_ERROR_CODE = "ZONE_RESOURCE_POOL_EXHAUSTED";
     static final String PREEMPTED_INSTANCE = "TERMINATED";
 
     private final Logger LOGGER = LoggerFactory.getLogger(ComputeEngine.class);
 
-    private final Arguments arguments;
+    private final CommonArguments arguments;
     private final Compute compute;
     private final Consumer<List<Zone>> zoneRandomizer;
     private final InstanceLifecycleManager lifecycleManager;
     private final BucketCompletionWatcher bucketWatcher;
+    private final boolean isShallow;
```

Comment:
Having the shallow as a flag on the compute engine is a problem. I see its just to propagate to the labels. We could invert the dependency and have the constructing code pass a label strategy.... but! No one uses the shallow label so I think we can just remove it